src/theory/strings/theory_strings.cpp

```diff
@@ -766,23 +776,36 @@ Node TheoryStrings::mkSkeletonFromBase(Node r,
                                        size_t currIndex,
                                        size_t nextIndex)
 {
+  Assert(nextIndex > currIndex);
   Assert(!r.isNull());
-  Assert(r.getType().isSequence());
   NodeManager* nm = NodeManager::currentNM();
   SkolemManager* sm = nm->getSkolemManager();
-  std::vector<Node> cacheVals;
-  cacheVals.push_back(r);
+  TypeNode tn = r.getType();
   std::vector<Node> skChildren;
-  TypeNode etn = r.getType().getSequenceElementType();
-  for (size_t i = currIndex; i < nextIndex; i++)
+  if (tn.isSequence())
   {
-    cacheVals.push_back(nm->mkConstInt(Rational(currIndex)));
-    Node kv = sm->mkSkolemFunction(
-        SkolemFunId::SEQ_MODEL_BASE_ELEMENT, etn, cacheVals);
-    skChildren.push_back(nm->mkNode(SEQ_UNIT, kv));
-    cacheVals.pop_back();
+    std::vector<Node> cacheVals;
+    cacheVals.push_back(r);
+    TypeNode etn = tn.getSequenceElementType();
+    for (size_t i = currIndex; i < nextIndex; i++)
+    {
+      cacheVals.push_back(nm->mkConstInt(Rational(currIndex)));
+      Node kv = sm->mkSkolemFunction(
+          SkolemFunId::SEQ_MODEL_BASE_ELEMENT, etn, cacheVals);
+      skChildren.push_back(utils::mkUnit(tn, kv));
+      cacheVals.pop_back();
+    }
   }
-  return utils::mkConcat(skChildren, r.getType());
+  else
+  {
+    // allocate a unique symbolic (unspecified) string of length one, and
+    // repeat it (nextIndex-currIndex) times.
+    d_strGapModelCounter++;
+    Node symChar =
+        utils::mkUnit(tn, nm->mkConstInt(-Rational(d_strGapModelCounter)));
```

Comment:
As discussed offline, it would be good to add a comment here to explain why this creates unique symbolic strings.